test/jdk/javax/imageio/plugins/tiff/compressions/TIFFCompressionsTest.java

```diff
@@ -0,0 +1,126 @@
+ * @summary Verify valid color space x compression combinations
+ */
+public class TIFFCompressionsTest {
+    public static void main(String[] args) throws IOException {
+        // @formatter:off
+        final String[][] testData = new String[][] {
+               // Filename                       OK/NOK  Regexp to match toString (OK) or exception (NOK)
+               // Expected pixel values for [0,0], [1,0], [2,0]
+  new String[] { "test_4x4_cmyk_jpeg.tiff",     "OK", ".*Bits = 32 numComponents = 4 .*SimpleCMYK.* transparency = 1 .*alpha = false.*",
+                 "[255, 254, 254, 0]", "[255, 3, 2, 255]", "[255, 0, 0, 3]"},
+  new String[] { "test_4x4_cmyk_lzw.tiff",      "OK", ".*Bits = 32 numComponents = 4 .*SimpleCMYK.* transparency = 1 .*alpha = false.*",
+                 "[0, 1, 0, 255]",     "[1, 254, 252, 1]", "[0, 255, 255, 249]"},
+  new String[] { "test_4x4_gray_jpeg.tiff",     "OK", ".*Bits = 8 numComponents = 1 .*ICC_.* transparency = 1 .*alpha = false.*",
+                 "[0]",                "[58]",             "[0]"},
+  new String[] { "test_4x4_grayalpha_lzw.tiff", "OK", ".*Bits = 16 numComponents = 2 .*ICC_.* transparency = 3 .*alpha = true.*",
+                 "[0, 254]",           "[54, 255]",        "[1, 127]"},
+  new String[] { "test_4x4_rgba_deflate.tiff",  "OK", ".*Bits = 32 numComponents = 4 .*ICC_.* transparency = 3 .*alpha = true.*",
+                 "[0, 0, 0, 254]",     "[254, 0, 0, 255]", "[2, 0, 0, 127]"},
+  new String[] { "test_4x4_rgba_lzw.tiff",      "OK", ".*Bits = 32 numComponents = 4 .*ICC_.* transparency = 3 .*alpha = true.*",
+                 "[0, 0, 0, 254]",     "[254, 0, 0, 255]", "[2, 0, 0, 127]"},
+  new String[] { "test_4x4_rgba_none.tiff",     "OK", ".*Bits = 32 numComponents = 4 .*ICC_.* transparency = 3 .*alpha = true.*",
+                 "[0, 0, 0, 254]",     "[254, 0, 0, 255]", "[2, 0, 0, 127]"},
+  new String[] { "test_4x4_rgba_packbits.tiff", "OK", ".*Bits = 32 numComponents = 4 .*ICC_.* transparency = 3 .*alpha = true.*",
+                 "[0, 0, 0, 254]",     "[254, 0, 0, 255]", "[2, 0, 0, 127]"},
+  new String[] { "test_4x4_rgba_jpeg.tiff",     "NOK","JPEG compressed tiles with alpha channel are not supported." },
+  new String[] { "test_4x4_grayalpha_jpeg.tiff","NOK","JPEG compressed tiles with alpha channel are not supported." },
+        };
+        // @formatter:on
+        final List<String> errors = new ArrayList<>();
+        for (String[] data : testData) {
+            final String filename = data[0].trim();
+            System.out.printf("Processing image %s.\n", filename);
+            final boolean ok = "OK".equals(data[1].trim());
+            final Pattern pattern = Pattern.compile(data[2].trim());
+            BufferedImage image = null;
+            String exceptionMsg = null;
+            try {
+                image = ImageIO.read(Objects.requireNonNull(TIFFCompressionsTest.class.getResource(filename)).openStream());
+            } catch (IIOException e) {
```

Comment:
Did you run this test inside jtreg ?
The test sources including the image files are generally NOT in the same place as the compiled class files.
ie the class files aren't compiled into the source directory.

Reply:
Thanks @prrace for looking at it.

This is how I ran the test:
```
$ make run-test TEST="jtreg:test/jdk/javax/imageio/plugins/tiff/compressions"
```
Getting the log location `/home/karm/workspaceRH/jdk/build/linux-x86_64-server-release/test-results/jtreg_test_jdk_javax_imageio_plugins_tiff_compressions/html/report.html` from the output:

```
Test selection 'jtreg:test/jdk/javax/imageio/plugins/tiff/compressions', will run:
* jtreg:test/jdk/javax/imageio/plugins/tiff/compressions

Running test 'jtreg:test/jdk/javax/imageio/plugins/tiff/compressions'
Passed: javax/imageio/plugins/tiff/compressions/TIFFCompressionsTest.java
Test results: passed: 1
Report written to /home/karm/workspaceRH/jdk/build/linux-x86_64-server-release/test-results/jtreg_test_jdk_javax_imageio_plugins_tiff_compressions/html/report.html
Results written to /home/karm/workspaceRH/jdk/build/linux-x86_64-server-release/test-support/jtreg_test_jdk_javax_imageio_plugins_tiff_compressions
Finished running test 'jtreg:test/jdk/javax/imageio/plugins/tiff/compressions'
Test report is stored in build/linux-x86_64-server-release/test-results/jtreg_test_jdk_javax_imageio_plugins_tiff_compressions

==============================
Test summary
==============================
   TEST                                              TOTAL  PASS  FAIL ERROR   
   jtreg:test/jdk/javax/imageio/plugins/tiff/compressions
                                                         1     1     0     0   
==============================
TEST SUCCESS
```

What is a better way of storing those resources? It seemed to me that there are various resources scattered throughout the testsuite. I am in the dark as to what is the actual best practice though.